### DISCUSSION TITLE
Welcome to lang-segment-anything Discussions!

### DISCUSSION THREAD

**Discussion by luca-medeiros** [luca-medeiros/lang-segment-anything#74] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Xamexer** [comment]
Dear Luca Medeiros,

I am currently studying at a university in Germany, where I am working on my Bachelor's thesis. My research focuses on evaluating text-to-segmentation models in the context of wildlife animal trap cameras. I have tested several models, including GroundedSAM2 and your LangSAM.

I found that LangSAM is by far one of the fastest models and delivers the best results. However, I was a bit confused when I noticed that it uses GroundedDINO and SAM 2.1, just like GroundedSAM2. This made me wonder why LangSAM still performs significantly better than GroundedSAM2.

Since there is no official paper on LangSAM, but I would like to include it in my thesis due to its impressive results, I initially reached out via LinkedIn and now through GitHub. Would it be possible to connect via LinkedIn or Discord to discuss the exact differences between GroundedSAM2 and LangSAM? This information would be extremely helpful for my research.

Thank you very much in advance!

Best regards,
Peter Rieser

**Reply by luca-medeiros** [reply]
Hi @Xamexer, I appreciate you checking out the project and considering using it in your thesis.

Currently, I don't have any benchmarks comparing Lang-Sam to other projects. My intuition is that there shouldn't be much difference in performance, as the base models are the same (unless they are using the closed version of grounding dino). Would you mind sharing your test results? 
Maybe we can come up with benchmarks that you could use in your thesis in a separate Discussion.

Let me know; I can help you with that.

**Reply by Xamexer** [reply]
Hello @luca-medeiros, thank you for the quick response.

To summarize, I created a small dataset of around 50 wildlife trap camera images (there isnt an available dataset with ground truth mask, so I made a small one for this project myself) with ground-truth masks (via Photoshop). I then used the same prompt to test different text-to-segmentation models and compiled the results in an Excel file. The models I tested include EVF-SAM, HyperSeg, LangSAM (yours), SEEM, and Grounding-SAM2—all without any prior training, just using the out-of-the-box pretrained versions.

Through this comparison, I found that LangSAM produced the best results, which made me curious as to why that is—especially since it doesn’t seem to use more modules than Grounding-SAM2. Both models use Grounding DINO and SAM2.1 with the same checkpoints. So why are the results so different? Why does LangSAM process images slightly faster and detect animals with a higher probability?
![2025-02-06 12-37-52](https://github.com/user-attachments/assets/cbb6c2af-0657-4f5e-8179-598c64978179)

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/Musa/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/Musa/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">


Modell | found object | avg s per image | mIOU (box) | IOU (mask)
-- | -- | -- | -- | --
EVF-SAM | 0,933 | 2,508 | 0,74 | 0,769
HyperSeg | 0,967 | 70,46 | 0,873 | 0,914
LangSAM | 0,967 | 0,569 | 0,907 | 0,929
SEEM | 0,733 | 0,241 | 0,702 | 0,759
Grounding-SAM   2.1 | 0,9 | 0,674 | 0,903 | 0,927



</body>

</html>

Thank you so much, cant wait for your response!

**Reply by luca-medeiros** [reply]
It’s great to hear that LangSAM performed well in your experiments.

Regarding the differences in performance, I would love to understand more about your testing methodology before drawing any conclusions. While your results are interesting, a few key factors could influence the outcomes:

1. Pipeline & Parameters: Different models might have different recommended postprocessing steps (e.g., mask refinement, confidence thresholds). Did all models use the same postprocessing approach, or was it handled differently?
1. Hardware & Environment: Were all models tested on the same GPU/CPU under identical conditions? Even slight variations in execution environments (e.g., different batch sizes, CUDA optimizations) can impact speed and performance.
1. Ground Truth Consistency: Since the ground truth masks were manually created, how were segmentation accuracy and thresholds determined? Minor annotation differences could affect IOU calculations.

Since it's all open source, you can check the differences and draw your conclusions. Also seems that grounded-sam2 uses DINO-X which is different from the model used in langsam.

Best,
Luca
